src/vs/workbench/api/common/extHostNotebook.ts

```diff
@@ -372,6 +378,113 @@ export class ExtHostNotebookController implements ExtHostNotebookShape {
 		return fileStats;
 	}
 
+	/**
+	 * Search for query in all notebooks that can be deserialized by the serializer fetched by `handle`.
+	 *
+	 * @param handle used to get notebook serializer
+	 * @param filenamePattern the filename pattern to match files that can be deserialized
+	 * @param textQuery the text query to search using
+	 * @param token cancellation token
+	 * @returns `IRawClosedNotebookFileMatch` for every file. Files without matches will just have a `IRawClosedNotebookFileMatch`
+	 * 	with no `cellResults`. This allows the caller to know what was searched in already, even if it did not yield results.
+	 */
+	async $searchInNotebooks(handle: number, filenamePattern: string[], textQuery: ITextQuery, token: CancellationToken): Promise<{ results: IRawClosedNotebookFileMatch[]; limitHit: boolean }> {
+		const serializer = this._notebookSerializer.get(handle)?.serializer;
+		if (!serializer) {
+			return {
+				limitHit: false,
+				results: []
+			};
+		}
+
+		const runFileQueries = async (includes: (string)[], token: CancellationToken, textQuery: ITextQuery): Promise<URI[]> => {
+			const uris = new ResourceSet();
+			await Promise.all(includes.map(include => {
+				const query: IFileQuery = {
+					type: QueryType.File,
+					filePattern: include,
+					folderQueries: textQuery.folderQueries,
+					maxResults: textQuery.maxResults,
+				};
+				return this._extHostSearch.$doInternalFileSearchWithCustomCallback(query, token, (data) => {
+					data.forEach(e => {
+						uris.add(URI.revive(e));
+					});
+				});
+			}));
+
+			return Array.from(uris.keys());
+		};
+
+		const filesToScan = await runFileQueries(filenamePattern, token, textQuery);
+
+		const results = new ResourceMap<INotebookFileMatchNoModel>();
+		let limitHit = false;
+		const promises = filesToScan.map(async (uri) => {
+			const cellMatches: INotebookCellMatchNoModel[] = [];
+
+			try {
+				if (token.isCancellationRequested) {
+					return;
+				}
+				if (textQuery.maxResults && [...results.values()].reduce((acc, value) => acc + value.cellResults.length, 0) > textQuery.maxResults) {
+					limitHit = true;
+					return;
+				}
+				const notebook = await this._notebookDataCache.getNotebookData(uri, async (data: VSBuffer) => {
+					return await this.$dataToNotebook(handle, data, token);
+				});
+				const cells = notebook.cells;
+
+				if (token.isCancellationRequested) {
+					return;
+				}
+
+				cells.forEach((cell, index) => {
+					const target = textQuery.contentPattern.pattern;
+					const cellModel: ICellSearchModel = cell instanceof NotebookCellTextModel ? new CellSearchModel('', cell.textBuffer, cell.outputs.flatMap(value => value.outputs)) : new CellSearchModel(cell.source, undefined, cell.outputs.flatMap(value => value.outputs));
```

Comment:
In EH we don't have `NotebookCellTextModel`, cell is always an object literal (`dto`) but not a concrete type, so this is always `false`

Reply:
oops, this is left over from the previous code